### ISSUE TITLE
LtUnlintNonExistingDefect falsely warns for range +unlint that covers only part of the defects

### ISSUE THREAD

**Issue by VasilevNStas** [objectionary/lints#1372]
## Problem

`LtUnlintNonExistingDefect` produces a false **"doesn't make sense"** warning when a line-range `+unlint` covers only *part* of the defects of a lint.

## Root cause

`src/main/java/org/eolang/lints/DefectMissing.java:50` uses `allMatch`:

```java
missing = !lines.stream().allMatch(new UnlintInRange(unlint));
```

For `+unlint foo:10-20` with `foo` defects on lines `[5, 15]`, `allMatch` returns `false` → the whole unlint is reported as "doesn't make sense", even though line `15` is legitimately suppressed by the range.

## Minimal example

EO program with `+unlint ascii-only:1-10` where the `ascii-only` lint reports defects both inside and outside the range (e.g. two non-ASCII comments on lines 5 and 15).

Expected: no warning — line 5 is suppressed by the range, line 15 is not and is reported by the lint itself.

Actual: `unlint-non-existing-defect` warns *"Suppressing 'ascii-only:1-10' doesn't make sense, because there are no defects with it"* (well — a defect exists, on line 15, but the warning fires anyway).

## Expected behavior

A range unlint makes sense if **any** defect of that lint falls into the range → use `anyMatch` (the same semantics `LtUnlint` uses via `removeIf(inRange)`, see `src/main/java/org/eolang/lints/LtUnlint.java:63`).

## Related

- #1232/#950 (line-specific `+unlint` handling)
- The pack `unlint/allows-unlint-for-defects-in-the-line-range.yaml` only covers the all-inside case, so the partial case is untested.

**Comment by 0crat** [comment]
@vasilevnstas Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+674](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
